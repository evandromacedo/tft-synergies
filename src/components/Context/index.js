import React, { useContext, useMemo, createContext } from 'react';
import useSynergies from '../../store';

const DispatchContext = createContext(null);
const StoreContext = createContext(null);

export default function Context({ children }) {
  const { state, ...dispatch } = useSynergies();
  const stateValue = useMemo(() => state, [state]);

  return (
    <DispatchContext.Provider value={{ ...dispatch }}>
      <StoreContext.Provider value={stateValue}>{children}</StoreContext.Provider>
    </DispatchContext.Provider>
  );
}

export const useDispatch = () => useContext(DispatchContext);
export const useStore = () => useContext(StoreContext);
