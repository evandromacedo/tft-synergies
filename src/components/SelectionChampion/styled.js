import styled from 'styled-components';
import Gold from './Gold.png';

export const Wrapper = styled.div`
  display: grid;
  grid-gap: 0.8rem;
  width: 12rem;
`;

export const Title = styled.h1`
  ${props => props.theme.bold20px}
  text-align: center;
`;

export const ImageAndCost = styled.div`
  height: 12rem;
  width: calc(100% - 2px);
  margin: 0 auto;
  background-image: url(${props => props.background});
  background-color: white;
  background-size: cover;
  box-shadow: inset -4px -4px 8px 2px black, inset 4px 4px 8px 2px black;
`;

export const Cost = styled.p`
  ${props => props.theme.bold20px}
  text-align: right;
  text-shadow: 0 0 6px black;
  padding: 0.8rem;
  position: relative;

  /* Gold Icon */
  &:before {
    content: '';
    position: absolute;
    top: 0.5rem;
    right: 2rem;
    width: 2.7rem;
    height: 2.3rem;
    background-image: url(${Gold});
    background-size: cover;
  }
`;
