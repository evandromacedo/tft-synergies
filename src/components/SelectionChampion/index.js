import React from 'react';
import * as S from './styled';
import TierBorder from '../Common/TierBorder';
import SynergyIcon from '../SynergyIcon';

export default function SelectionChampion() {
  const getImgSrc = function(name) {
    return `https://rerollcdn.com/characters/${name}.png`;
  };

  return (
    <S.Wrapper>
      <S.Title>Mordekaiser</S.Title>
      <TierBorder cost={5}>
        <S.ImageAndCost background={getImgSrc('Kayle')}>
          <S.Cost>5</S.Cost>
        </S.ImageAndCost>
        <div>
          <SynergyIcon classOrOrigin="Demon" type="hover" />
        </div>
      </TierBorder>
    </S.Wrapper>
  );
}
