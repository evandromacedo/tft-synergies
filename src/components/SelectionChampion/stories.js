import React from 'react';
import SelectionChampion from '.';
import { storiesOf } from '@storybook/react';

storiesOf('SelectionChampion', module)
  .add('Camille', () => <SelectionChampion champion={camille} />)
  .add('Blitzcrank', () => <SelectionChampion champion={blitzcrank} />)
  .add('Kennen', () => <SelectionChampion champion={kennen} />)
  .add('Gnar', () => <SelectionChampion champion={gnar} />)
  .add('Kayle', () => <SelectionChampion champion={kayle} />);

const camille = {
  id: '164',
  key: 'Camille',
  name: 'Camille',
  origin: ['Hextech'],
  class: ['Blademaster'],
  cost: 1,
  ability: {
    name: 'The Hextech Ultimatum',
    description:
      "Camille singles out an enemy, dealing magic damage and rooting them for few seconds. Camille's allies in range will prioritize attacking that enemy.",
    type: 'Active',
    manaCost: 150,
    manaStart: 0,
    stats: [
      {
        type: 'Damage',
        value: '200 / 325 / 450'
      },
      {
        type: 'Root Duration',
        value: '4s / 5s / 6s'
      }
    ]
  },
  stats: {
    offense: {
      damage: 55,
      attackSpeed: 0.6,
      dps: 30,
      range: 1
    },
    defense: {
      health: 550,
      armor: 25,
      magicResist: 20
    }
  },
  items: ['spearofshojin', 'titanichydra', 'guardianangel']
};

const blitzcrank = {
  id: '53',
  key: 'Blitzcrank',
  name: 'Blitzcrank',
  origin: ['Robot'],
  class: ['Brawler'],
  cost: 2,
  ability: {
    name: 'Rocket Grab',
    description: 'Blitzcrank pulls the furthest enemy to him.',
    type: 'Active',
    manaCost: 125,
    manaStart: 0,
    stats: [
      {
        type: 'Damage',
        value: '100 / 350 / 800'
      },
      {
        type: 'Stun Duration',
        value: '2.5s'
      }
    ]
  },
  stats: {
    offense: {
      damage: 50,
      attackSpeed: 0.5,
      dps: 25,
      range: 1
    },
    defense: {
      health: 600,
      armor: 35,
      magicResist: 20
    }
  },
  items: ['thornmail', 'warmogsarmor', 'dragonsclaw']
};

const kennen = {
  id: '85',
  key: 'Kennen',
  name: 'Kennen',
  origin: ['Ninja', 'Yordle'],
  class: ['Elementalist'],
  cost: 3,
  ability: {
    name: 'Slicing Maelstrom',
    description:
      'Kennen summons a storm around him, dealing damage and stunning enemies inside of it.',
    type: 'Active',
    manaCost: 150,
    manaStart: 50,
    stats: [
      {
        type: 'Damage',
        value: '225 / 450 / 675'
      },
      {
        type: 'Stun Duration',
        value: '1.5s'
      }
    ]
  },
  stats: {
    offense: {
      damage: 65,
      attackSpeed: 0.65,
      dps: 36,
      range: 2
    },
    defense: {
      health: 550,
      armor: 20,
      magicResist: 20
    }
  },
  items: ['morellonomicon', 'guardianangel', 'hextechgunblade']
};

const gnar = {
  id: '150',
  key: 'Gnar',
  name: 'Gnar',
  origin: ['Wild', 'Yordle'],
  class: ['Shapeshifter'],
  cost: 4,
  ability: {
    name: 'GNAR!',
    description:
      'Gnar transforms and jumps behind the furthest enemy, damaging and shoving enemies backwards.',
    type: 'Active',
    manaCost: 100,
    manaStart: 0,
    stats: [
      {
        type: 'Damage',
        value: '200 / 300 / 400'
      },
      {
        type: 'CC Duration',
        value: '2s'
      },
      {
        type: 'Knockback Distance',
        value: '2 hexes'
      },
      {
        type: 'Transform Duration',
        value: '60s'
      },
      {
        type: 'Transform Bonus HP',
        value: '300 / 600 / 900'
      },
      {
        type: 'Transform Bonus Attack Damage',
        value: '50 / 100 / 150'
      }
    ]
  },
  stats: {
    offense: {
      damage: 50,
      attackSpeed: 0.7,
      dps: 31,
      range: 2
    },
    defense: {
      health: 750,
      armor: 30,
      magicResist: 20
    }
  },
  items: ['guardianangel', 'warmogsarmor', 'dragonsclaw']
};

const kayle = {
  id: '10',
  key: 'Kayle',
  name: 'Kayle',
  origin: ['Noble'],
  class: ['Knight'],
  cost: 5,
  ability: {
    name: 'Divine Judgement',
    description: 'Kayle shields an ally, making them immune to damage.',
    type: 'Active',
    manaCost: 125,
    manaStart: 0,
    stats: [
      {
        type: 'Shield Duration',
        value: '2s / 2.5s / 3s'
      },
      {
        type: 'Extra Targets',
        value: '0 / 1 / 2'
      }
    ]
  },
  stats: {
    offense: {
      damage: 60,
      attackSpeed: 1,
      dps: 66,
      range: 3
    },
    defense: {
      health: 750,
      armor: 40,
      magicResist: 20
    }
  },
  items: ['bladeoftheruinedking', 'guinsoosrageblade', 'spearofshojin']
};
