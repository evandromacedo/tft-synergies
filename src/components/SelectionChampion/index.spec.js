import React from 'react';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';
import { wrapInTestContext } from 'react-dnd-test-utils';
import * as Context from '../Context';
import SelectionChampion from '.';

const kayleMock = {
  id: '10',
  key: 'Kayle',
  name: 'Kayle',
  origin: ['Noble'],
  class: ['Knight'],
  cost: 5,
  ability: {
    name: 'Divine Judgement',
    description: 'Kayle shields an ally, making them immune to damage.',
    type: 'Active',
    manaCost: 125,
    manaStart: 0,
    stats: [
      {
        type: 'Shield Duration',
        value: '2s / 2.5s / 3s'
      },
      {
        type: 'Extra Targets',
        value: '0 / 1 / 2'
      }
    ]
  },
  stats: {
    offense: {
      damage: 60,
      attackSpeed: 1,
      dps: 66,
      range: 3
    },
    defense: {
      health: 750,
      armor: 40,
      magicResist: 20
    }
  },
  items: ['bladeoftheruinedking', 'guinsoosrageblade', 'spearofshojin']
};

jest.spyOn(Context, 'useDispatch').mockImplementation(() => ({ addChampion: () => {} }));

describe('<SelectionChampion />', () => {
  it('renders properly', () => {
    const SelectionChampionWrapped = wrapInTestContext(SelectionChampion);
    shallow(<SelectionChampionWrapped champion={kayleMock} />);
  });

  it('renders and matches snapshot', () => {
    const SelectionChampionWrapped = wrapInTestContext(SelectionChampion);
    const wrapper = shallow(<SelectionChampionWrapped champion={kayleMock} />);
    expect(toJSON(wrapper)).toMatchSnapshot();
  });
});
