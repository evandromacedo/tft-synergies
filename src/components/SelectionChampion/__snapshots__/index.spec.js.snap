// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SelectionChampion /> renders and matches snapshot 1`] = `
<Memo()
  backend={[Function]}
>
  <SelectionChampion
    champion={
      Object {
        "ability": Object {
          "description": "Kayle shields an ally, making them immune to damage.",
          "manaCost": 125,
          "manaStart": 0,
          "name": "Divine Judgement",
          "stats": Array [
            Object {
              "type": "Shield Duration",
              "value": "2s / 2.5s / 3s",
            },
            Object {
              "type": "Extra Targets",
              "value": "0 / 1 / 2",
            },
          ],
          "type": "Active",
        },
        "class": Array [
          "Knight",
        ],
        "cost": 5,
        "id": "10",
        "items": Array [
          "bladeoftheruinedking",
          "guinsoosrageblade",
          "spearofshojin",
        ],
        "key": "Kayle",
        "name": "Kayle",
        "origin": Array [
          "Noble",
        ],
        "stats": Object {
          "defense": Object {
            "armor": 40,
            "health": 750,
            "magicResist": 20,
          },
          "offense": Object {
            "attackSpeed": 1,
            "damage": 60,
            "dps": 66,
            "range": 3,
          },
        },
      }
    }
  />
</Memo()>
`;
